src/components/common/PaginationTable.tsx

```diff
@@ -0,0 +1,77 @@
+import { Loader } from "@ahaui/react";
+import classNames from "classnames";
+import React, { useEffect, useState } from "react";
+import { DataTable, TableColumnType } from "types/table";
+import Pagination from "./Pagination";
+import Table from "./Table";
+
+type PaginationTableProps = {
+  data: DataTable | undefined;
+  tableName: string;
+  cols: TableColumnType[];
+  fetchData: (offset: number) => void;
+  pageSize: number;
+  CreateButton: React.ReactNode;
+};
+
+const PaginationTable: React.FC<PaginationTableProps> = ({
+  data,
+  tableName,
+  cols,
+  fetchData,
+  pageSize,
+  CreateButton,
+}) => {
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```

Comment:
```suggestion
  const [currentPage, setCurrentPage] = useState(1);
  const [isLoading, setIsLoading] = useState(false);
```
No need to define type, it can infer from initial value.